codalab/worker/worker_run_state.py

```diff
@@ -264,10 +289,41 @@ def _transition_from_PREPARING(self, run_state):
                     self.dependency_manager.dependencies_dir,
                     self.dependency_manager.get(run_state.bundle.uuid, dep_key).path,
                 )
-                os.symlink(docker_dependency_path, full_child_path)
-            # These are turned into docker volume bindings like:
-            #   dependency_path:docker_dependency_path:ro
-            docker_dependencies.append((dependency_path, docker_dependency_path))
+
+                if dep.child_path != '.':
+                    # A child path can be a nested path which we need to create directories for
+                    Path(full_child_path).mkdir(parents=True, exist_ok=True)
+                    try:
```

Comment:
Just curious how we think about refactoring file operations from using `os` to `pathlib`?  `pathlib` seems to be more convenient and compact and easier to read. We might be able to remove a lot of extra code by using `pathlib`. If we think this is something worth doing, we could file a separate ticket?

Reply:
I prefer `pathlib` too. It provides more utility than `os` too.